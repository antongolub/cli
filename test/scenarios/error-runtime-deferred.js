'use strict'
require('../..')(main)

function defer () {
  return new Promise(r => setImmediate(r))
}

async function main () {
  console.log('started')
  await defer() 
  main.test.example = 'will explode'
  console.log('finished')
}
