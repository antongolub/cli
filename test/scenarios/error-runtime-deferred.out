started
TypeError: Cannot set property 'example' of undefined
    at main (scenarios/error-runtime-deferred.js:)
